### ISSUE TITLE
add doc_cfg on all `std` api's

### ISSUE THREAD

**Issue by yaahc** [yaahc/eyre#25]
Right now we have nominal support for `no_std` thanks to `anyhow`, but its not really clear which APIs are available and which aren't when using std vs no_std.

To help with this we should use the new `doc_cfg` feature to mark APIs that are and aren't available with std enabled. The tracing-subscriber repo is a good example to use as a reference for how to setup `doc_cfg`.

https://docs.rs/tracing-subscriber/

**Comment by brightly-salty** [comment]
I would be willing to attempt this and submit a PR if it would be accepted. It looks like just adding `#[cfg_attr(docsrs, doc(cfg(feature = "std")))]` to declarations with `#[cfg(feature = "std")]` ?

**Comment by yaahc** [comment]
I forgot to update this issue back when I changed things, but I don't think we even nominally support no_std mode anymore. As far as I can tell I removed all those `cfg`'s back when I switched from a generic parameter to store the `EyreHandler` to instead boxing a `dyn EyreHandler`. 

https://github.com/yaahc/eyre/commit/f39eaff0b2c8306dc86ca54e62bc12f6ff1e604e

The only other two features that we have, `track_caller` and `backtrace` don't affect our public API as far as I can tell, so `doc_cfg`s wouldn't help them at all. I'm really sorry about the confusion.
